### PR TITLE
Fix has_static_new

### PR DESCRIPTION
**Pull request by yeswalrus**
MSVC2015, alone among MSVC versions and for reasons that are beyond me chokes on the function name detection portion of the old has_static_new's SFINAE.
This overhauls it to be far more concise, and compile correctly.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -45,3 +45,5 @@ autowiring.kdev4
 autowiring.VC.db
 .vscode
 AutowiringVersion.h
+
+*.VC.db*
```
Comment (jdonald) [inline .gitignore]: `Autowiring.VC.db is already listed above though?

Or if the intent here is the trailing asterisk, are there circumstances when it makes a db1, db2 file?

File: src/autowiring/test/AutoConstructTest.cpp [hunk 2]
```diff
@@ -102,3 +107,65 @@ TEST_F(AutoConstructTest, FactoryNewPrivateCtor) {
...
+  "Found a bad static new"
+  );
+static_assert(
+  autowiring::has_static_new<MultiStaticNew, int>::value,
+  "Failed to find a good static new override"
+  );
+
+TEST_F(AutoConstructTest, MultiStaticNewTest) {
+  AutoConstruct<MultiStaticNew> badstatic{ 1 };
+}
```
Comment (jdonald) [inline src/autowiring/test/AutoConstructTest.cpp]: missing newline at EOF

Comment (jdonald) [inline src/autowiring/test/AutoConstructTest.cpp]: This var is named `badstatic`. However, it looks like the good one?

File: src/autowiring/has_static_new.h [hunk 3]
```diff
@@ -5,51 +5,18 @@
 
 namespace autowiring {
 
-/// <summary>
-/// Utility helper structure for types which have a factory New routine
-/// </summary>
-/// <remarks>
-/// A factory New routine is malformed when the return type is not implicitly castable to type T
-/// </remarks>
```
Comment (jdonald) [inline src/autowiring/has_static_new.h]: Should keep the comments? I found the `<remarks>` line useful initially.

**Comment by jdonald** [comment]
This implementation is effectively `-38 +10`. I've tested platform on OS X and MSVC2013 and it probably works everywhere. Great work, @yeswalrus !

File: .gitignore (hunk 1 above)
Comment (yeswalrus) [inline .gitignore]: Yeah, 2017 adds several extensions.

Comment (jdonald) [inline .gitignore]: I see now. I have `Browse.VC.db` and `Solution.VC.db{,-shm,-wal}`.

Should remove the now-redundant `autowiring.VC.db` above then?

File: .editorconfig [hunk 4]
```diff
@@ -0,0 +1,6 @@
+root = true
+
+[**]
+indent_style = space
+indent_size = 2
+insert_final_newline = true
```
Comment (jdonald) [inline .editorconfig]: Ironic that this file is missing its final newline!

Comment (yeswalrus) [inline .editorconfig]: Sadly, VS2017 has default support for editor config but ignores the 2 most useful options: insert_final_newline and trim_trailing_whitespace

File: .gitignore (hunk 1 above)
Comment (yeswalrus) [inline .gitignore]: yep

**Pull request merged**